### PR TITLE
Feat: Enable full SFN actions in api gateway integration

### PR DESCRIPTION
**Pull request by aazbeltran**
Currently, only the StartExecution is enabled in the API Gateway integration, I spent hours thinking that I was doing something wrong in my definition until I found that the error was in localstack, this PR adds dynamism in the method to be used, supporting any sfn action

### CODE REVIEW

**Comment by whummer** [comment]
LGTM, thanks for making the API GW -> SFN integration generic @aazbeltran . 👍  

Looks like we still have a minor test issue - can you please take a look if this is reprodicible locally?

```
>       self.assertEqual(200, resp.status_code)
E       AssertionError: 200 != 400

/opt/code/localstack/tests/integration/test_api_gateway.py:1122: AssertionError
```

You should be able to run a single test method with this command:
```
DEBUG=1 TEST_PATH=tests/integration/test_api_gateway.py::TestAPIGateway::test_step_function_integrations make test
```

File: tests/integration/test_api_gateway.py [hunk 1]
```diff
@@ -1193,17 +1192,24 @@ def _prepare_method_integration(
...
 
         # invoke stepfunction via API GW, assert results
         test_data_1["name"] += "2"
         resp = requests.post(url, data=json.dumps(test_data_1))
         self.assertEqual(200, resp.status_code)
         self.assertEqual(test_data, json.loads(to_str(resp.content.decode())))
 
+        _prepare_method_integration(overwrite=True, action="DeleteStateMachine")
+
+        # Remove state machine with API GW
```
Comment (whummer) [inline tests/integration/test_api_gateway.py:1207]: nit: should probably read `# Invoke ...`

File: tests/integration/test_api_gateway.py [hunk 2]
```diff
@@ -1193,17 +1192,24 @@ def _prepare_method_integration(
...
+        resp = requests.post(url, data=json.dumps({"stateMachineArn": sm_arn}))
+        self.assertEqual(200, resp.status_code)
+
         # Clean up
         lambda_client.delete_function(FunctionName=fn_name)
-        sfn_client.delete_state_machine(stateMachineArn=sm_arn)
```
Comment (whummer) [inline tests/integration/test_api_gateway.py:1206]: Out of curiosity - was there an issue with this call, any particular reason why it has been removed?

**Review by whummer — APPROVED** [review]
Awesome, thanks for updating the PR @aazbeltran . 👍  Two minor comments - shouldn't hold up the merge, though (we'll pick them up in a future iteration..) Thanks

File: tests/integration/test_api_gateway.py (hunk 2 above)
Comment (aazbeltran) [reply tests/integration/test_api_gateway.py:1206]: The state machine is deleted via API Gateway to test other integrations different to startexecution, clearly we cannot eliminate twice the same state machine

 https://github.com/localstack/localstack/pull/4448/files#diff-b0a7d5551e1c6d1516c3cbfb62572f4b70e45bbef9113560054a0cc702c1a886R1204-R1209

Comment (whummer) [reply tests/integration/test_api_gateway.py:1206]: Ah, of course - my bad. Thanks for clarifying!

**Pull request merged**